apps/designer/app/designer/features/topbar/menu/menu.tsx

```diff
@@ -26,6 +34,41 @@ const menuItemCss = {
   minWidth: 140,
 };
 
+const ThemeMenuItem = () => {
+  if (isFeatureEnabled("theme") === false) return null;
+  const currentSetting = getThemeSetting();
+  const labels: Record<ThemeSetting, string> = {
+    light: "Light",
+    dark: "Dark",
+    system: "System theme",
+  };
+
+  const settings = Object.keys(labels) as Array<ThemeSetting>;
+
+  return (
+    <DropdownMenu>
+      <DropdownMenuTriggerItem>
+        Theme
+        <ChevronRightIcon />
+      </DropdownMenuTriggerItem>
+      <DropdownMenuContent>
+        {settings.map((setting) => (
+          <DropdownMenuCheckboxItem
+            key={setting}
+            checked={currentSetting === setting}
+            css={menuItemCss}
+            onSelect={() => {
+              setThemeSetting(setting);
+            }}
+          >
+            {labels[setting]}
+          </DropdownMenuCheckboxItem>
+        ))}
+      </DropdownMenuContent>
+    </DropdownMenu>
```

Comment:
Consider using `Select` as it should simplify the usage a lot.

Reply:
That doesn't work, its an actual menu with submenus

Reply:
Ah I see, it's a sub-menu in another menu.